libs/wingc/src/type_check/jsii_importer.rs

```diff
@@ -756,27 +757,56 @@ impl<'a> JsiiImporter<'a> {
 	}
 
 	fn parameter_to_wing_type(&mut self, parameter: &jsii::Parameter) -> TypeRef {
+		let mut param_type = self.type_ref_to_wing_type(&parameter.type_);
+
 		if parameter.variadic.unwrap_or(false) {
-			panic!("TODO: variadic parameters are unsupported - Give a +1 to this issue: https://github.com/winglang/wing/issues/397");
+			// TODO properly thought out variadic parameter support https://github.com/winglang/wing/issues/397
+			param_type = self.wing_types.add_type(Type::Array(param_type));
+			param_type = self.wing_types.add_type(Type::Optional(param_type));
 		}
```

Comment:
We now eagerly load types in the root of imported jsii packages. In the root of "constructs", one of the types has a variadic param.

I feel like this approach is okay, but I didn't want to close the issue because I didn't really think that hard about it. If this change is sufficient then I'll close it.

Reply:
I think the only sure-fire way to know if this works (now and going forward) is to add an e2e test where a variadic param is used. 🙂

The main thing we want to avoid is e.g. code-generating `variadicFunction([1, 2, 3])` if, say, the underlying JavaScript expects `variadicFunction(1, 2, 3)`.

(I'm totally cool with this workaround and leaving the issue open if removing this panic is needed for LSP stuff to work)

Reply:
Okay this still straight up doesn't work during preflight. I made it more clear in the comment but still kept the workaround in-place